File: GeeksforGeeks_POTD/July_2026/July_9.py
"""
Make pairs for remainder
And for middle and rem=0--> act differently

TC - O(n), SC - O(k)
"""

class Solution:
    def countKdivPairs(self, arr, k):
        rem=[0]*k
        for i in arr:
            rem[i%k]+=1
        ans=(rem[0]*(rem[0]-1))//2;
        l=1
        h=k-1
        while l<h:
            ans+=rem[l]*rem[h]
            l+=1
            h-=1
        if l==h: ans+=(rem[l]*(rem[l]-1))//2;
        return ans

File: GeeksforGeeks_POTD/July_2026/July_9.java
package GeeksforGeeks_POTD.July_2026;

/*
Make pairs for remainder
And for middle and rem=0--> act differently

TC - O(n), SC - O(k)
*/

public class July_9 {
    public int countKdivPairs(int[] arr, int k) {
        int rem[]=new int[k];
        for(int i:arr){
            rem[i%k]++;
        }
        int ans=(rem[0]*(rem[0]-1))/2;
        int l=1, h=k-1;
        while(l<h){
            ans+=rem[l]*rem[h];
            l++; h--;
        }
        if(l==h){
            ans+=(rem[l]*(rem[l]-1))/2;
        }
        return ans;
    }
}
